src/generate_worker_metadata.py

```diff
@@ -0,0 +1,142 @@
+
+
+@task
+def generate_worker_metadata(worker_subcls: Type[BaseWorker], package_name: str):
+    """Generates worker metadata."""
+
+    worker_metadata = dict(
+        sorted(
+            {
+                "type": worker_subcls.type,
+                "install_command": f"pip install {package_name}",
+                "description": worker_subcls.get_description(),
+                "logo_url": worker_subcls.get_logo_url(),
+                "documentation_url": worker_subcls.get_documentation_url(),
+                "default_base_job_configuration": worker_subcls.get_default_base_job_configuration(),
+            }.items()
+        )
+    )
+
+    validate(worker_metadata, worker_schema)
+
+    return worker_metadata
+
+
+@flow
+def get_worker_metadata_from_prefect():
+    """Gets worker metadata from prefect."""
+
+    worker_registry = get_registry_for_type(BaseWorker) or {}
+
+    output = {
+        "workers": {
+            "prefect-agent": {
+                "type": "prefect-agent",
+                "install_command": "pip install prefect",
+                "default_base_job_configuration": {},
+                "description": "A Prefect agent that executes flow runs via infrastructure blocks.",
+            }
+        }
+    }
+
+    metadata = {
+        worker_subcls.type: generate_worker_metadata(
+            worker_subcls=worker_subcls, package_name="prefect"
+        )
+        for worker_subcls in worker_registry.values()
+    }
+
+    output["workers"].update(metadata)
+    output["workers"] = dict(sorted(output["workers"].items()))
+    return output
+
+
+@flow
+def get_worker_metadata_from_collection(collection_name: str):
+    """Gets worker metadata from a given collection."""
+    collections = safe_load_entrypoints(entry_points(group="prefect.collections"))
+
+    output = {"workers": {}}
+    for ep_name, module in collections.items():
+        if isinstance(module, Exception):
+            logging.warning(f"Error loading collection entrypoint {ep_name} - skipping")
+            continue
+        discovered_collection_name = module.__name__.split(".")[0].replace("_", "-")
+        if collection_name != discovered_collection_name:
+            continue
+
+        worker_subclasses = discover_base_worker_subclasses(module)
+        for worker_subcls in worker_subclasses:
+            output["workers"].update(
+                generate_worker_metadata(
+                    worker_subcls=worker_subcls, package_name=collection_name
+                )
+            )
+
+    output["workers"] = dict(sorted(output["workers"].items()))
+    return output
+
+
+@task
+def discover_base_worker_subclasses(module: ModuleType) -> List[Type[BaseWorker]]:
+    return [
+        cls
+        for _, cls in inspect.getmembers(module)
+        if inspect.isclass(cls)
+        and issubclass(cls, BaseWorker)
+        and cls.__name__ != "BaseWorker"
+    ]
+
+
+@task
+def write_worker_metadata(worker_metadata: Dict[str, Any], package_name: str):
```

Comment:
curious why we need to write to files? if we were, would we need to add a github action for this?

i guess im wondering if we can just pass the output from `generate_worker_metadata_for_package` to `submit_updates`

Reply:
This was mostly for local development. I assume our production flow would use `generate_worker_metadata_for_package` in conjunction with `submit_updates` like you suggested.